src/main/java/com/epam/spring/homework2/validator/impl/BeanValidatorImpl.java

```diff
@@ -0,0 +1,20 @@
+package com.epam.spring.homework2.validator.impl;
+
+import com.epam.spring.homework2.validator.BeanValidator;
+
+public class BeanValidatorImpl implements BeanValidator {
+
+    public static void validate(String className, String name, int value) {
+        if (name != null && value > 0) {
+            System.out.println(className + " successfully validated");
+        } else {
+            System.out.println(className + " unsuccessfully validated");
+        }
+    }
+
+    @Override
+    public void validate() {
+
+    }
+}
```

Comment:
Overall, validation logic looks good, just a small remark - it would be good to return some value from the method despite having the validation result logged so that consumers of that method could do something if the validation failed - throw an Exception, return error status code, etc.

However, class design could be improved - having a single static method in the class, which would validate class instance or methods to validate each field would be better. Then, those methods could be called inside the post processor. What do you think about that?

Reply:
I did not figure out how to get class instances in order to extract data from them for validation in the postprocessor, so I decided to add my own annotation and work with it .  I hope this will also be the right solution, but if anything, I will be happy to think about other options for solving this problem. I corrected the rest of the comments, they were very useful.

Reply:
Using marker annotation is great solution! Other way around could be one abstract class for each bean, which would contain field which need to be validated. I have also seen Visitor pattern being used to make the validation. Those approaches are less resource-consuming than reflection, but your implementation is also pretty good)